src/Simulation/Simulators.Impl/QCTraceSimulator/QCTraceSimulator.Checks.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Runtime.InteropServices;
+using Microsoft.Quantum.Simulation.Core;
+
+namespace Microsoft.Quantum.Simulation.Simulators.QCTraceSimulators.Implementation
+{
+    public partial class QCTraceSimulatorImpl
```

Comment:
This was done to make the APIs for the target packages behave like the existing ones for the negative tests (things like passing the same qubit in the controls array as the target). We discussed possibly changing the tests instead of exposing this validation logic in Q#, and I'm going to investigate that as part of the updates we discussed.

Reply:
I was able to remove the CheckQubitUniqueness and RotationAngleValidation helpers.